client/my-sites/domains/domain-search/index.jsx

```diff
@@ -123,7 +123,7 @@ class DomainSearch extends Component {
 
 		addItem( registration );
 
-		if ( ! isGSuiteRestricted() && canDomainAddGSuite( domain ) ) {
+		if ( canDomainAddGSuite( domain ) ) {
```

Comment:
There is no need to call `isGSuiteRestricted()` since `canDomainAddGSuite()` actually takes care of that.